server/src/model/family.py

```diff
@@ -22,16 +23,17 @@ class Family(Document):
     @property
     def mates(self) -> List[Mate]:
         """Fetch object for mate_ids."""
-        return [mate.fetch() for mate in self.mate_ids]
+        return list(Mate.objects.in_bulk([ref.id for ref in self.mate_ids]).values())
 
     def get_id(self) -> str:
         """Return string version of mongo oid."""
         return str(self.id)
 
     @staticmethod
     @error_bounded(None)
-    def lookup_family(id: str) -> Optional[Family]:
+    def lookup_family(family: Union[str, LazyReference]) -> Optional[Family]:
```

Comment:
Allow for `LazyReference`

Reply:
This is painful to look at, but I guess it lets the syntax on the caller side make sense. I'm now questioning whether we should get rid of the `_id` on all the LazyRef fields.
Would be a diff ticket tho.

Reply:
unfortunately agreed